modules/porous_flow/src/userobjects/PorousFlowDictator.C

```diff
@@ -0,0 +1,120 @@
+  params.addRequiredParam<unsigned int>("number_fluid_phases", "The number of fluid phases in the simulation");
+  params.addRequiredParam<unsigned int>("number_fluid_components", "The number of fluid components in the simulation");
+  return params;
+}
+
+PorousFlowDictator::PorousFlowDictator(const InputParameters & parameters) :
+    GeneralUserObject(parameters),
+    Coupleable(this, false),
+    ZeroInterface(parameters),
+    _num_variables(coupledComponents("porous_flow_vars")),
+    _num_phases(getParam<unsigned int>("number_fluid_phases")),
+    _num_components(getParam<unsigned int>("number_fluid_components"))
+{
+  unsigned int max_moose_var_num_seen = 0;
+
+  _moose_var_num.resize(_num_variables);
+  for (unsigned int i = 0; i < _num_variables; ++i)
+  {
+    _moose_var_num[i] = coupled("porous_flow_vars", i);
+    max_moose_var_num_seen = (max_moose_var_num_seen > _moose_var_num[i] ? max_moose_var_num_seen : _moose_var_num[i]);
+  }
+
+  _pf_var_num.resize(max_moose_var_num_seen + 1);
+  for (unsigned int i = 0; i < max_moose_var_num_seen + 1; ++i)
+    _pf_var_num[i] = _num_variables; // NOTE: indicates that i is not a PorousFlow variable
+  for (unsigned int i = 0; i < _num_variables; ++i)
+    _pf_var_num[_moose_var_num[i]] = i;
+}
+
+void
+PorousFlowDictator::initialize()
+{}
+
+void
+PorousFlowDictator::execute()
+{}
+
+void
+PorousFlowDictator::finalize()
+{}
+
+unsigned int
+PorousFlowDictator::numVariables() const
+{
+  return _num_variables;
+}
+
+unsigned int
+PorousFlowDictator::numPhases() const
+{
+  return _num_phases;
+}
+
+unsigned int
+PorousFlowDictator::numComponents() const
+{
+  return _num_components;
+}
+
+unsigned int
+PorousFlowDictator::porousFlowVariableNum(unsigned int moose_var_num) const
+{
+  if (moose_var_num >= _pf_var_num.size() || _pf_var_num[moose_var_num] == _num_variables)
+    mooseError("The Dictator proclaims that the moose variable with number " << moose_var_num << " is not a PorousFlow variable.  Exiting with error code 1984.");
+  return _pf_var_num[moose_var_num];
+}
+
+bool
+PorousFlowDictator::isPorousFlowVariable(unsigned int moose_var_num) const
+{
+  return !notPorousFlowVariable(moose_var_num);
+}
+
+bool
+PorousFlowDictator::notPorousFlowVariable(unsigned int moose_var_num) const
+{
+  if (moose_var_num >= _pf_var_num.size() || _pf_var_num[moose_var_num] == _num_variables)
+    return true;
+  return false;
+}
+
+const VariableName
+PorousFlowDictator::pressureVariableNameDummy() const
+{
+  return "pressure_variable_dummy";
+}
+
+const VariableName
+PorousFlowDictator::saturationVariableNameDummy() const
+{
+  return "saturation_variable_dummy";
+}
+
+const VariableName
+PorousFlowDictator::temperatureVariableNameDummy() const
+{
+  return "temperature_variable_dummy";
+}
+
+const VariableName
+PorousFlowDictator::massFractionVariableNameDummy() const
+{
+  return "mass_fraction_variable_dummy";
```

Comment:
I'm with @dschwen here. If there's a deficiency in MOOSE, let's address that. We've seen developers do this before to get around some requirement or assumption in MOOSE. Later, we've simply removed that assumption and made all of that code unecessary.